spec/lib/web_of_science/query_author_spec.rb

```diff
@@ -0,0 +1,121 @@
+# http://savonrb.com/version2/testing.html
+# require the helper module
+require 'savon/mock/spec_helper'
+
+describe WebOfScience::QueryAuthor do
+  include Savon::SpecHelper
+
+  # set Savon in and out of mock mode
+  before(:all) { savon.mock!   }
+  after(:all)  { savon.unmock! }
+
+  subject(:query_author) { described_class.new(author) }
+
+  let(:names) { query_author.send(:names) }
+  let(:institution) { query_author.send(:institution) }
+
+  # These wos_uids are in the wos_search_author_query_response.xml
+  let(:wos_uids) { %w[MEDLINE:29218869 WOS:000422723300003] }
+  let(:wos_search_author_query_response) { File.read('spec/fixtures/wos_client/wos_search_author_query_response.xml') }
+
+  let(:author) do
+    # public data from
+    # - https://stanfordwho.stanford.edu
+    # - https://med.stanford.edu/profiles/russ-altman
+    author = FactoryBot.create(:author,
+                                 preferred_first_name: 'Russ',
+                                 preferred_last_name: 'Altman',
+                                 preferred_middle_name: 'Biagio',
+                                 email: 'Russ.Altman@stanford.edu',
+                                 cap_import_enabled: true)
+    # create some `author.alternative_identities`
+    FactoryBot.create(:author_identity,
+                       author: author,
+                       first_name: 'R',
+                       middle_name: 'B',
+                       last_name: 'Altman',
+                       email: nil,
+                       institution: 'Stanford University')
+    FactoryBot.create(:author_identity,
+                       author: author,
+                       first_name: 'Russ',
+                       middle_name: nil,
+                       last_name: 'Altman',
+                       email: nil,
+                       institution: nil)
+    author
+  end
+
+  it 'works' do
+    expect(query_author).to be_a described_class
+  end
+
+  describe '#uids' do
+    let(:wos_auth_response) { File.read('spec/fixtures/wos_client/authenticate.xml') }
+
+    before do
+      wos_client = WebOfScience::Client.new('secret')
+      allow(WebOfScience).to receive(:client).and_return(wos_client)
+      savon.expects(:authenticate).returns(wos_auth_response)
+      savon.expects(:search).with(message: :any).returns(wos_search_author_query_response)
+    end
+
+    it 'returns an Array<String> of WOS-UIDs' do
+      expect(query_author.uids).to eq wos_uids
+    end
+  end
+
+  # PRIVATE
+
+  describe '#author_query' do
+    let(:query) { query_author.send(:author_query) }
+    let(:params) { query[:queryParameters] }
+
+    it 'contains query parameters' do
+      expect(query).to include(queryParameters: Hash)
+      expect(params).to include(databaseId: String, userQuery: String, timeSpan: Hash, queryLanguage: String)
+    end
+
+    it 'contains author name' do
+      expect(params[:userQuery]).to include(names)
```

Comment:
These 3 tests can be combined into a single `includes` expectation.

Reply:
I don't care to make this change, it's fine as it is.  I prefer to get back more specific information about what is failing.  Since the userQuery is a String with various moving parts, separate specs are useful.